app/components/account-settings/details/component.js

```diff
@@ -0,0 +1,118 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class AccountSettingsDetailsComponent extends Component {
+  @service store;
+  @tracked editMode = false;
+
+  get currentUser() {
+    return this.args.currentUser;
+  }
+
+  get shouldBeButtonDisabled() {
+    return !this.currentUser.hasDirtyAttributes;
+  }
+
+  get shouldBeLocked() {
+    return !this.editMode;
+  }
+
+  get isFormFilledCorrectly() {
+    const { currentUser } = this;
+    const isRequiredInputsFilled = Boolean(
+      currentUser.name &&
+        currentUser.surname &&
+        currentUser.username &&
+        currentUser.password &&
+        currentUser.email
+    );
+    return isRequiredInputsFilled;
+  }
+
+  @action
+  onChangeName({ target: { value } }) {
+    this.currentUser.name = value;
+  }
+
+  @action
+  onChangeSurname({ target: { value } }) {
+    this.currentUser.surname = value;
+  }
+
+  @action
+  onChangeUsername({ target: { value } }) {
+    this.currentUser.username = value;
+  }
+
+  @action
+  onChangePassword({ target: { value } }) {
+    this.currentUser.password = value;
+  }
+
+  @action
+  onChangeEmail({ target: { value } }) {
+    if (value === '') {
+      this.currentUser.email = null;
+    } else {
+      this.currentUser.email = value;
+    }
+  }
```

Comment:
Option 1:
```suggestion
  @action
  onChangeEmail({ target: { value } }) {
    if (value === '') {
      return this.currentUser.email = null;
    }
    
    this.currentUser.email = value;
  }
```

Option 1:
```suggestion
  @action
  onChangeEmail({ target: { value } }) {
    this.currentUser.email = value || null;
  }
```